src/hll.c

```diff
@@ -92,6 +109,308 @@ enum {
+	}
+
+	aggregateValuesInitialized = true;
+}
+
+/*
+ * get_extension_schema - given an extension OID, fetch its extnamespace
+ * Returns InvalidOid if no such extension.
+ *
+ * Copied from postgresql extension.c.
+ */
+static Oid
+get_extension_schema(Oid ext_oid)
+{
+	Oid			result;
+	Relation	rel;
+	SysScanDesc scandesc;
+	HeapTuple	tuple;
+	ScanKeyData entry[1];
+
+	rel = heap_open(ExtensionRelationId, AccessShareLock);
+
+	ScanKeyInit(&entry[0],
+				ObjectIdAttributeNumber,
+				BTEqualStrategyNumber, F_OIDEQ,
+				ObjectIdGetDatum(ext_oid));
+
+	scandesc = systable_beginscan(rel, ExtensionOidIndexId, true,
+								  NULL, 1, entry);
+
+	tuple = systable_getnext(scandesc);
+
+	/* We assume that there can be at most one matching tuple */
+	if (HeapTupleIsValid(tuple))
+		result = ((Form_pg_extension) GETSTRUCT(tuple))->extnamespace;
+	else
+		result = InvalidOid;
+
+	systable_endscan(scandesc);
+
+	heap_close(rel, AccessShareLock);
+
+	return result;
+}
+
+/*
+ * FunctionOid searches for a given function identified by schema, functionName
+ * and argumentCount. It reports error if the function is not found or there
+ * are more than one match. If the missingOK parameter is set and there are
+ * no matches, then the function returns InvalidOid.
+ */
+static Oid
+FunctionOid(const char *schemaName, const char *functionName, int argumentCount,
+					bool missingOK)
+{
+	FuncCandidateList functionList = NULL;
+	Oid functionOid = InvalidOid;
+
+	char *qualifiedFunctionName = quote_qualified_identifier(schemaName, functionName);
+	List *qualifiedFunctionNameList = stringToQualifiedNameList(qualifiedFunctionName);
+	List *argumentList = NIL;
+	const bool findVariadics = false;
+	const bool findDefaults = false;
+
+	functionList = FuncnameGetCandidates(qualifiedFunctionNameList, argumentCount,
+										 argumentList, findVariadics,
+										 findDefaults, true);
+
+	if (functionList == NULL)
+	{
+		if (missingOK)
+		{
+			return InvalidOid;
+		}
+
+		ereport(ERROR, (errcode(ERRCODE_UNDEFINED_FUNCTION),
+						errmsg("function \"%s\" does not exist", functionName)));
+	}
+	else if (functionList->next != NULL)
+	{
+		ereport(ERROR, (errcode(ERRCODE_AMBIGUOUS_FUNCTION),
+						errmsg("more than one function named \"%s\"", functionName)));
+	}
+
+	/* get function oid from function list's head */
+	functionOid = functionList->oid;
+
+	return functionOid;
+}
+
+/*
+ * MaximizeCostOfHashAggregate maximizes the cost of the path if it tries
+ * to run hll aggregate function with hash aggregate.
+ */
+static void
+MaximizeCostOfHashAggregate(Path *path)
+{
+	List *varList = pull_var_clause((Node*) path->pathtarget->exprs,
+									PVC_INCLUDE_AGGREGATES);
+	ListCell *varCell = NULL;
+
+	foreach(varCell, varList)
+	{
+		Var *var = (Var *) lfirst(varCell);
+
+		if (nodeTag(var) == T_Aggref)
+		{
+			Aggref *aggref = (Aggref *) var;
+
+			if(HllAggregateOid(aggref->aggfnoid))
+			{
+				path->total_cost = INT_MAX;
```

Comment:
INT_MAX is a pretty arbitrary value. 

`total_cost` is not an int. If it was then setting it to the max would be a bad idea because it would likely overflow when other costs are added, now it's just arbitrary, which is probably fine?